src/util.ts

```diff
@@ -47,3 +51,32 @@ export function getFileExtension(pathOrUrl: string): string {
         pathOrUrl
     );
 }
+
+export function isBinarySimulariumFile(fileContents: Blob): Promise<boolean> {
+    const first16blob: Blob = fileContents.slice(0, 16);
+    // compare this with "SIMULARIUMBINARY"
+    return first16blob.text().then((text) => {
+        return text === "SIMULARIUMBINARY";
+    });
+}
+
+export function loadSimulariumFile(file: Blob): Promise<ISimulariumFile> {
+    return isBinarySimulariumFile(file)
+        .then((isBinary): Promise<ArrayBuffer | string> => {
+            if (isBinary) {
+                return file.arrayBuffer();
+            } else {
+                return file.text();
+            }
+        })
+        .then((fileContents: ArrayBuffer | string) => {
+            if (typeof fileContents === "string") {
+                console.log("TEXT JSON FILE");
```

Comment:
Don't we have a logger for this sort of stuff?

Reply:
That was for my own debugging really... I'll get rid of it.